### DISCUSSION TITLE
Is there any support from Drop wizard to integrate with rate limiting the API's

### DISCUSSION THREAD

**Discussion by kumaramit06** [dropwizard/dropwizard#4797] [General]
We are looking for rate-limiting our APIs at the user level. We are using drop wizard as one of the components in our application. Is there any POC or sample rate limiting implementation around the community ? Please help us with this. Thanks a lot in advance.

**Comment by zUniQueX** [comment]
If you want rate limiting for a DoS protection, you could take a look at Jetty's [DoSFilter](http://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/servlets/DoSFilter.html)

**Comment by shahr** [comment]
I'd recommend taking a look at https://github.com/vladimir-bukhtoyarov/bucket4j. We've integrated this very successfully into our dropwizard based service. It works really well. (I'm not the owner or affiliated with the library)

**Comment by daghendrik** [comment]
@shahr : Do you by any chance know of any example code or tutorial on how to integrate bucket4j with dropwizard?

**Comment by zUniQueX** [comment]
@daghendrik the documentation provides a link to a community [Dropwizard integration](https://github.com/zivver/dropwizard-ratelimit). This however uses a very old Dropwizard version. But you may get a starting point for integrating it to the current version.

**Comment by zUniQueX** [comment]
@kumaramit06 did this work out for you? Then I'd request you to close this issue

**Comment by zUniQueX** [comment]
@joschi As we now have the nice feature of discussions, I think this would be a better place for this than an issue :wink:
